engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -217,6 +217,7 @@
   protected String shutdownDiagnosticsMessage = "";
   private long lastResourceRequest = 0;
   private final Map<String, StreamingContainerAgent> containers = new ConcurrentHashMap<>();
+  private final Set<String> containerHeartBeatMissSet = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
```

Comment:
Why is the Set necessary? Will not it be easier to track that the container time outed on the container itself?